camayoc/tests/qpc/api/v1/reports/test_reports.py

```diff
@@ -100,10 +99,9 @@ def test_report_content_consistency(data_provider):
     validate_json_response(report.deployments(headers=accept_json))
 
 
-@pytest.mark.skip("Skipped until Middleware are handled by Camayoc")
 @pytest.mark.runs_scan
 @pytest.mark.parametrize("scan_info", scan_list(), ids=utils.name_getter)
```

Comment:
`scan_list()` will return all scans that are of type network, vcenter, satellite. But can we actually detect any middleware on vcenter and satellite?

If not, I would suggest to create new `network_scan_list()` function, that would return network scans only (or maybe `scan_list` could take a param?). This should save some time by not running scans that are irrelevant in this context anyway.

Even if we can detect middlewares on Satellite and VCenter I would consider not running these scans, because our test lab does not have middlewares on them. But that would require a short explanation in comment. Something along the lines of "we can detect middleware on Satellite and VCenter, but we don't have this in our test lab, so we skip them to save some pipeline time".

Reply:
Well, I prefered to update `scan_list()` with a custom iterable for filtering source types.